src/Xamarin.Android.Build.Tasks/Tasks/ExtractJarsFromAar.cs

```
@@ -0,0 +1,93 @@
+				var annotations = new HashSet<string> (StringComparer.OrdinalIgnoreCase);
+				foreach (var library in Libraries) {
+					bool isAar = library.EndsWith (".aar", StringComparison.OrdinalIgnoreCase);
+					var jarOutputDirectory = Path.Combine (OutputJarsDirectory, Path.GetFileName (library));
+					var annotationOutputDirectory = Path.Combine (OutputAnnotationsDirectory, Path.GetFileName (library));
+					using (var zip = MonoAndroidHelper.ReadZipFile (library)) {
+						foreach (var entry in zip) {
+							if (entry.IsDirectory)
+								continue;
+							var entryFullName = entry.FullName;
+							var fileName = Path.GetFileName (entryFullName);
+							if (string.Equals (fileName, "annotations.zip", StringComparison.OrdinalIgnoreCase)) {
+								var path = Path.GetFullPath (Path.Combine (annotationOutputDirectory, entryFullName));
+								Extract (entry, memoryStream, path);
+								annotations.Add (path);
+							} else if (!entryFullName.EndsWith (".jar", StringComparison.OrdinalIgnoreCase)) {
+								continue;
+							} else if (isAar && Files.ShouldSkipEntryInAar (entryFullName)) {
+								continue;
+							} else {
+								var path = Path.GetFullPath (Path.Combine (jarOutputDirectory, entryFullName));
+								Extract (entry, memoryStream, path);
+								jars.Add (path);
+							}
+						}
+					}
+				}
+				DeleteUnknownFiles (OutputJarsDirectory, jars);
+				DeleteUnknownFiles (OutputAnnotationsDirectory, annotations);
+			} finally {
+				MemoryStreamPool.Shared.Return (memoryStream);
+			}
+
+			return !Log.HasLoggedErrors;
+		}
+
+		static void Extract (ZipEntry entry, MemoryStream stream, string destination)
+		{
+			stream.SetLength (0); //Reuse the stream
+			entry.Extract (stream);
+			try {
+				MonoAndroidHelper.CopyIfStreamChanged (stream, destination);
+			} catch (PathTooLongException) {
+				throw new PathTooLongException ($"Could not extract \"{entry.FullName}\" to \"{destination}\". Path is too long.");
```

Why are we wrapping a `PathToLongException` with a `PathToLongException`? Assuming we need to, shouldn't we "preserve" the nested exception?

Furthermore, can't this possibly wind up in build error output?  Does this message need to be localized?

---

This was copied from another task, but yeah it seems like I should just let this throw.